app/src/main/java/com/realityexpander/tasky/agenda_feature/data/repositories/agendaRepository/agendaRepositoryImpls/AgendaRepositoryImpl.kt

```diff
@@ -0,0 +1,308 @@
+package com.realityexpander.tasky.agenda_feature.data.repositories.agendaRepository.agendaRepositoryImpls
+
+import com.realityexpander.remindery.agenda_feature.data.common.convertersDTOEntityDomain.toDomain
+import com.realityexpander.tasky.agenda_feature.common.util.EventId
+import com.realityexpander.tasky.agenda_feature.common.util.ReminderId
+import com.realityexpander.tasky.agenda_feature.common.util.TaskId
+import com.realityexpander.tasky.agenda_feature.data.common.convertersDTOEntityDomain.toDomain
+import com.realityexpander.tasky.agenda_feature.data.repositories.agendaRepository.remote.IAgendaApi
+import com.realityexpander.tasky.agenda_feature.data.repositories.attendeeRepository.IAttendeeRepository
+import com.realityexpander.tasky.agenda_feature.data.repositories.syncRepository.ISyncRepository
+import com.realityexpander.tasky.agenda_feature.data.repositories.syncRepository.local.AgendaItemTypeForSync
+import com.realityexpander.tasky.agenda_feature.data.repositories.syncRepository.local.ModificationTypeForSync
+import com.realityexpander.tasky.agenda_feature.domain.*
+import com.realityexpander.tasky.auth_feature.domain.IAuthRepository
+import com.realityexpander.tasky.core.util.Email
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.combine
+import kotlinx.coroutines.flow.emitAll
+import kotlinx.coroutines.flow.flow
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.supervisorScope
+import java.time.ZonedDateTime
+import javax.inject.Inject
+
+class AgendaRepositoryImpl @Inject constructor(
+    private val agendaApi: IAgendaApi,
+    private val eventRepository: IEventRepository,
+    private val attendeeRepository: IAttendeeRepository,
+    private val reminderRepository: IReminderRepository,
+    private val taskRepository: ITaskRepository,
+    private val syncRepository: ISyncRepository,
+    private val authRepository: IAuthRepository
+) : IAgendaRepository {
+
+    ///////////////////////////////////////////////
+    // • AGENDA
+
+    override suspend fun getAgendaForDay(dateTime: ZonedDateTime): List<AgendaItem> {
+        val events = eventRepository.getEventsForDay(dateTime)
+        val tasks = taskRepository.getTasksForDay(dateTime)
+        val reminders = reminderRepository.getRemindersForDay(dateTime)
```

Comment:
These calls are all independent of each other, so they can be executed in parallel:
```
return supervisorScope {
    val events = async { eventRepository.getEventsForDay(dateTime) }
    // ...
    events.await() + tasks.await() + reminders.await()
}
```

Reply:
Those are calls to DAO's and only return the local items.

## FOLLOW UP QUESTION 3

I thought that because the database calls are already coroutines and accessing the same database, that there is no need to parallelize their calls.

Do database calls need to be parallelized?

Reply:
Yes because in your example the call to `taskRepository.getTasksForDay(dateTime)` will only be executed when the previous event retrieval finished. All that is executed in the same suspend function, so it gets executed sequentially. Internally, DBs will surely do a lot of multi-threading but only for one given call. Here your DAO can't know that there'll be a future call to getTasks

Reply:
OK, didnt know you could do this!

I will refactor